### PR TITLE
Adding breadcumbs for Dashboards

### PR DESCRIPTION
**Pull request by npfitz**
Epic #22826 

Small change for AppBar Breadcrumbs to be displayed in Dashboards

image

### CODE REVIEW

File: frontend/src/metabase/dashboard/containers/DashboardApp.jsx [hunk 1]
```diff
@@ -120,6 +133,15 @@ const DashboardApp = props => {
     onTimeout,
   });
 
+  const collectionId = dashboard?.collection_id;
+
+  useEffect(() => {
+    if (collectionId) {
+      setCollectionId(collectionId);
+      return () => clearBreadcrumbs();
+    }
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/containers/DashboardApp.jsx:142]: From my testing, this doesn't show the breadcrumb when the dashboard is on a `root` collection aka `Our analytics`. I don't think this is the intent as we also show the exact same breadcrumb for questions in the root collection.

This is how the question call `setCollectionId`

https://github.com/metabase/metabase/blob/70ba4d5c73022af1d5113a29429a3534dbfce67e/frontend/src/metabase/query_builder/containers/QueryBuilder.jsx#L261-L266

Not sure if we really need to check if `collectionId` exists, but remove the if clause makes the breadcrumb appear for dashboard on the root collection.

```suggestion
    setCollectionId(collectionId);
    return () => clearBreadcrumbs();
```

File: frontend/src/metabase/dashboard/containers/DashboardApp.jsx [hunk 2]
```diff
@@ -18,7 +18,12 @@ import { useWebNotification } from "metabase/hooks/use-web-notification";
 import { useOnUnmount } from "metabase/hooks/use-on-unmount";
 
 import { fetchDatabaseMetadata } from "metabase/redux/metadata";
-import { getIsNavbarOpen, setErrorPage } from "metabase/redux/app";
+import {
+  getIsNavbarOpen,
+  setErrorPage,
+  setCollectionId,
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/containers/DashboardApp.jsx:24]: I believe it was incorrect to store `collectionId` in the redux state in the first place. We should have defined it as a selector. Same for `showBreadcrumbs`. Can this be changed in this PR?

**Review by ranquild — CHANGES_REQUESTED** [review]
A comment about `collectionId`. Could be fixed later.

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/dashboard/containers/DashboardApp.jsx (hunk 2 above)
Comment (npfitz) [reply frontend/src/metabase/dashboard/containers/DashboardApp.jsx:24]: We can certainly make a change here, and I expect it to change quite a bit when we handle the "forked question" use case as well. The Dashboard looks like it resets part of it's state when you navigate away from the page. I can also do something similar in the query builder

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/selectors/app.ts [hunk 3]
```diff
@@ -1,11 +1,15 @@
 import { Location } from "history";
 import { createSelector } from "reselect";
 import { getUser } from "metabase/selectors/user";
-import { getIsEditing as getIsEditingDashboard } from "metabase/dashboard/selectors";
+import {
+  getIsEditing as getIsEditingDashboard,
```
[the remark this answers is not in the document]
Comment (npfitz) [reply frontend/src/metabase/selectors/app.ts:5]: `getIsEditingDashboard` is used by a couple selectors that existed before (`getIsNavBarVisible`)

**Pull request merged**